### ISSUE TITLE
Inconsistent file name prefix for saving model and loading model

### ISSUE THREAD

**Issue by zorksylar** [dmlc/mxnet#3645]
For example, when using `mxnet/example/image-classification/train_model.py` for training, and set `args.save_model_prefix` with `hello`, the checkpoint file name is `hello-<epoch_number>.param` and `hello-symbol.json`. 
But when loading model with `args.model_prefix=hello`, mxnet will find for file with prefix `hello-<kv.rank>`. 

Error log : 

```
python train_cifar10.py --gpus=0 --data-dir=/scratch1/pingao/data/cifar/ --batch-size=256
 --lr=0.08 --model-prefix=/home/pingao/dev-pla/mxnet/example/image-classification/models/models-batch/cifar10-b512 --load-e
poch=10 --num-epochs=2 --log_file=cifar10-switch-b512-e10-b256.log
[04:12:20] include/dmlc/logging.h:235: [04:12:20] src/io/local_filesys.cc:154: Check failed: allow_null  LocalFileSystem: f
ail to open "/home/pingao/dev-pla/mxnet/example/image-classification/models/models-batch/cifar10-b512-0-symbol.json"
Traceback (most recent call last):
  File "train_cifar10.py", line 85, in <module>
    train_model.fit(args, net, get_iterator)
  File "/home/pingao/dev-pla/mxnet/example/image-classification/train_model.py", line 37, in fit
    tmp = mx.model.FeedForward.load(model_prefix, args.load_epoch)
  File "/home/pingao/dev-pla/mxnet/python/mxnet/model.py", line 837, in load
    symbol, arg_params, aux_params = load_checkpoint(prefix, epoch)
  File "/home/pingao/dev-pla/mxnet/python/mxnet/model.py", line 362, in load_checkpoint
    symbol = sym.load('%s-symbol.json' % prefix)
  File "/home/pingao/dev-pla/mxnet/python/mxnet/symbol.py", line 958, in load
    check_call(_LIB.MXSymbolCreateFromFile(c_str(fname), ctypes.byref(handle)))
  File "/home/pingao/dev-pla/mxnet/python/mxnet/base.py", line 77, in check_call
    raise MXNetError(py_str(_LIB.MXGetLastError()))
mxnet.base.MXNetError: [04:12:20] src/io/local_filesys.cc:154: Check failed: allow_null  LocalFileSystem: fail to open "/ho
me/pingao/dev-pla/mxnet/example/image-classification/models/models-batch/cifar10-b512-0-symbol.json"
```

Is this a bug or I just missed something.

**Comment by VoVAllen** [comment]
add "--num-epochs <your epochs>" to the command may could solve this problem

**Comment by zorksylar** [comment]
Can be fixed by PR #3648

**Comment by zorksylar** [comment]
This issue is fixed by the latest commit by #3999
